mobile/src/controllers/NotificationsController.ts

```diff
@@ -254,6 +268,22 @@ export class NotificationsController implements IDisposable {
         this.settings.updateNotifications({ enabled: this._notificationsEnabledByUser }, 'enabled');
     };
 
+    public goToAffirmation(): boolean {
+        const now = new Date();
+
+        const lastOpen = this.settings.current.notifications.lastAffirmationOpenDate;
+        if ((now.getTime() - lastOpen) > 86400000) {
+            if (!this.scheduleTime) {
+                this.scheduleTime = this.settings.current.notifications.scheduledTime;
+            }
+
+            let affirmationTime = new Date();
+            affirmationTime.setHours(this.scheduleTime.hour, this.scheduleTime.minute, 0);
+            return now > affirmationTime;
+        }
+        return false;
```

Comment:
Given that we want to record the ACTUAL affirmation opened time, I think this check should have this logic:

> Show the affirmation screen IF the user last viewed an affirmation before today's notification was sent.

More specifically, this translates to:
> Show the affirmation screen IF the lastAffirmationOpenDate is before today's scheduledTime.

... which would be something along the lines of:
```
if (!this.scheduleTime) {
    this.scheduleTime = this.settings.current.notifications.scheduledTime;
}
const notificationSentDate = new Date();
notificationSentDate.setHours(this.scheduleTime.hour, this.scheduleTime.minute, 0, 0);

return lastAffirmationOpenDate < notificationSentDate;
```